### DISCUSSION TITLE
[alpinejs compnents] x-dialog closes on mouseup

### DISCUSSION THREAD

**Discussion by moxx** [alpinejs/alpine#4230] [7. Bugs (archived)]
if you have a mousedown action in your modal which influences the content height and you then suddenly come out of the modal with the mouse click "mousedown" then the modal closes because the "mouseup" takes place outside the modal. 
The problem is that you cannot influence this behavior, so there is no option to deactivate the click outside in general, or to bind it to mousedown only.

https://codepen.io/moxxs/pen/wvbWvrY

**Comment by SimoTod** [comment]
I don't think it's a bug, it's just the way it works. 
It's a standard browser event that originates from outside the modal, Alpine does have much control on it.
Can't you put the action on mouseup? 
Usually with buttons, you never trigger an action until you let the mouse button go.

**Comment by moxx** [comment]
If its not a bug than its a feature,. I cant change the mousedown because i use x-tabs and that is how they behave. I think i cant change this, can use my own tab switcher.
But I would very much welcome being able to control this a little more.

**Reply by SimoTod** [reply]
> If its not a bug than its a feature

I meant that it's just the obvious way it would work with plain javascript regardless of Alpine.

I would say that it's more an usability issue with x-tab them. it should listen for a mouse up rather then a mouse down.
That was changed in Tailwind UI as well: https://github.com/tailwindlabs/headlessui/issues/845

You can probably make a PR to change the tab component and pitch it to the maintainer mentioning that tailwind thread.

**Reply by moxx** [reply]
Where can I do a PR for alpinejs components?

**Reply by ekwoka** [reply]
https://github.com/alpinejs/alpine

**Reply by SimoTod** [reply]
@moxx FYI, I've just drafted a PR to change that: https://github.com/alpinejs/alpine/pull/4239

**Reply by moxx** [reply]
Thank you very much 🙂
